File: src/main/java/com/hirehub/SpringServer/Entity/ApplicationStatusLog.java
package com.hirehub.SpringServer.Entity;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.sql.Timestamp;

@Entity
@Table(name = "applicationStatusLogs")
@Data
@NoArgsConstructor
@AllArgsConstructor
public class ApplicationStatusLog {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long logId;

    @ManyToOne
    @JoinColumn(name = "applicationId")
    private Application application;

    private String fromStatus;
    private String toStatus;

    private Timestamp updatedAt;
}

File: DotNetServer/Models/Entities/ApplicationStatusLog.cs
using System.ComponentModel.DataAnnotations;

namespace HireHubServer.Models.Entities;

public class ApplicationStatusLog
{
  [Key]
  public long LogId { get; set; }

  public long ApplicationId { get; set; }

  public string? FromStatus { get; set; }
  public string? ToStatus { get; set; }

  public DateTime UpdatedAt { get; set; }

  public Application Application { get; set; } = null!;
}
